src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetVersionService.cs

```diff
@@ -182,12 +182,41 @@ public async Task<Either<ActionResult, DataSetDraftVersionViewModel>> UpdateVers
             .OnSuccess(async dataSetVersion => await MapDraftVersion(dataSetVersion, cancellationToken));
     }
 
+    public async Task UpdateVersionsForReleaseVersion(
+        Guid releaseVersionId,
+        string slug,
+        string title,
+        CancellationToken cancellationToken = default)
+    {
+        var releaseFileIds = await contentDbContext
+            .ReleaseFiles
+            .Where(rf => rf.ReleaseVersionId == releaseVersionId && rf.File.Type == FileType.Data)
+            .Select(rf => rf.Id)
+            .ToListAsync(cancellationToken);
+
+        var dataSetVersions = await publicDataDbContext
+            .DataSetVersions
+            .Where(dataSetVersion => releaseFileIds.Contains(dataSetVersion.Release.ReleaseFileId))
+            .ToListAsync(cancellationToken);
+
+        if (dataSetVersions.Count > 0)
+        {
+            foreach (var dsv in dataSetVersions)
+            {
+                dsv.Release.Slug = slug;
+                dsv.Release.Title = title;
+            }
+
+            await publicDataDbContext.SaveChangesAsync(cancellationToken);
+        }
+    }
+
     private async Task<IReadOnlyDictionary<Guid, ReleaseVersion>> GetReleasesVersionsByDataSetVersionId(
         IReadOnlyList<DataSetVersion> dataSetVersions,
         CancellationToken cancellationToken)
     {
         var dataSetVersionsByReleaseFileId = dataSetVersions
-            .ToDictionary(dsv => dsv.ReleaseFileId);
+            .ToDictionary(dsv => dsv.Release.ReleaseFileId);
```

Comment:
I would have expected this to blow up because we didn't do a `.Include(dsv => dsv.Release)` when grabbing the `DataSetVersion` which is passed into this method

Reply:
You already messaged me with the answer in a DM but just to resolve this comment, the answer is here:

[Querying owned types](https://learn.microsoft.com/en-us/ef/core/modeling/owned-entities#querying-owned-types)

> When querying the owner the owned types will be included by default. It is not necessary to use the Include method, even if the owned types are stored in a separate table.